kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/Environment.java

```diff
@@ -33,7 +37,21 @@ private Environment() {
     /**
      * The kroxy version default value
      */
-    public static final String KROXY_VERSION_DEFAULT = "0.5.0-SNAPSHOT";
+    public static final String KROXY_VERSION_DEFAULT;
+
+    static {
+        var p = new Properties();
+        try (var stream = Environment.class.getResourceAsStream("/metadata.properties");) {
+            Objects.requireNonNull(stream, "/metadata.properties absent");
+            p.load(stream);
+            var version = p.getProperty("kroxylicious.version");
+            Objects.requireNonNull(version, "kroxylicious version key absent in metadata.properties");
```

Comment:
Maybe it would be good to fail fast if the placeholder hadn't been replaced. We could check it doesn't start with`$` as a simple check. You could also do a more fiddly check that it's a valid tag then you'd get an error like `"${project.version}" is not a valid tag`.